src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,406 @@
+            {
+                int newBegin = max(detectBegin - (windowLength - 1), 0);
+                detectEnd += detectBegin - newBegin;
+                detectBegin = newBegin;
+            }
+            beginTensor[id_z] = detectBegin;
+            lengthTensor[id_z] = detectEnd;
+        }
+    }
+}
+
+// -------------------- Set 4 -  host helpers for kernel executor --------------------
+
+int prev_pow2(int n)
+{
+  int pow2 = 1;
+  while (n - pow2 > pow2)
+    pow2 += pow2;
+
+  return pow2;
+}
+
+int next_pow2(int n)
+{
+  int pow2 = 1;
+  while (n > pow2)
+    pow2 += pow2;
+
+  return pow2;
+}
+
+// -------------------- Set 5 - non silent region kernels executor --------------------
+
+RppStatus hip_exec_non_silent_region_detection_tensor(Rpp32f *srcPtr,
+                                                      RpptDescPtr srcDescPtr,
+                                                      Rpp32s *srcLengthTensor,
+                                                      Rpp32s *detectedIndexTensor,
+                                                      Rpp32s *detectionLengthTensor,
+                                                      Rpp32f cutOffDB,
+                                                      Rpp32s windowLength,
+                                                      Rpp32f referencePower,
+                                                      Rpp32s resetInterval,
+                                                      rpp::Handle& handle)
+{
+    // allocate temporary memory for MMS Array
+    Rpp32f *mmsArr;
+    hipMalloc(&(mmsArr), srcDescPtr->n * srcDescPtr->strides.nStride * sizeof(Rpp32f));
+
+    int maxSharedMemoryInBytes = 32000; // 32 KB
+    int maxSharedMemoryElements = maxSharedMemoryInBytes / sizeof(Rpp32f);
+    int kSharedMemBanks = 32;
+    int inputTileLength = prev_pow2(maxSharedMemoryElements * kSharedMemBanks / (kSharedMemBanks + 1));
+
+    if (resetInterval > 0 && resetInterval < inputTileLength)
+    {
+        auto p = prev_pow2(resetInterval);
+        auto n = next_pow2(resetInterval);
+        if (p > windowLength)
+            inputTileLength = p;
+        else if (n < inputTileLength)
+            inputTileLength = n;
+    }
+
+    int sharedMemorySizeInBytes = smem_pos(inputTileLength) * sizeof(Rpp32f);
+    int outputTileLength = inputTileLength - windowLength;
+    float windowFactor = 1.0f / windowLength;
+
+    if (outputTileLength <= 0)
+    {
+        std::cout << "Invalid output tile length! " << std::endl;
+        return RPP_ERROR;
+    }
+    if (sharedMemorySizeInBytes > maxSharedMemoryInBytes)
+    {
+        std::cout << "Cannot compute the requested moving mean square, due to shared memory restrictions" << std::endl;
+        return RPP_ERROR;
+    }
+
+    // launch kernel to compute the values needed for MMS Array
+    int globalThreads_x = std::min<int>(ceil(static_cast<float>(srcDescPtr->strides.nStride) / outputTileLength), 1024);
```

Comment:
How was global x set? What if nStride / outputTileLength is well beyond 1024? We should discuss on a call.